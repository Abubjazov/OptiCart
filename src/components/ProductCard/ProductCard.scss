@import '../../variables';

.product {
	height: 340px;
	width: 200px;
	background-color: $white;

	@media (max-width: 436px) {
		height: 200px;
		width: 350px;
		background-color: $gray;

		display: grid;
		grid-template-columns: 200px 1fr;
		gap: 5px;
		grid-template-rows: 40px 1fr;
		grid-template-areas:
			'image descriptor'
			'image footer';
	}

	img {
		grid-area: image;
		height: 200px;
		width: 200px;
	}

	.product-descriptor {
		grid-area: descriptor;
		padding: 0 10px;
		margin-top: 10px;
		height: 60px;
		line-height: 18px;
		font-size: 14px;
		font-weight: 300;
		color: $black;

		@media (max-width: 436px) {
			background-color: $white;
			height: 40px;
			margin: 0;
			padding: 7px 0;
		}

		h2 {
			line-height: 25px;
			font-size: 25px;
			font-weight: 300;
			color: $main-color;
			font-weight: 300;
			margin-bottom: 5px;

			@media (max-width: 436px) {
				text-align: center;
			}
		}

		p {
			@media (max-width: 436px) {
				display: none;
			}
		}
	}

	footer {
		grid-area: footer;
		display: flex;
		justify-content: space-between;
		margin-top: 30px;
		margin-left: 13px;
		margin-right: 5px;
		background: $white;
		color: $black;
		font-size: 17px;
		line-height: 30px;
		font-weight: 700;

		@media (max-width: 436px) {
			flex-direction: column;
			text-align: center;
			margin: 0;
			padding-top: 40px;
			font-size: 25px;
			line-height: 21px;
			font-weight: 300;
		}

		button {
			width: 85px;
			height: 30px;
			border: none;
			padding: 0 5px;
			background-color: $main-color;
			color: $white;
			font-size: 15px;
			line-height: 15px;
			cursor: pointer;
			transition: all 0.2s linear;

			@media (max-width: 436px) {
				width: 145px;
				height: 50px;
				font-size: 19px;
				line-height: 19px;
			}

			&:hover {
				background-color: $deep-main;
			}
		}
	}
}
