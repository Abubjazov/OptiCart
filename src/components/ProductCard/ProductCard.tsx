import { useActions } from '../../hooks/useActions'
import { useTypedSelector } from '../../hooks/useTypedSelector'

import { Product } from '../../interfaces/product.interface'
import { SmallSpinner } from '../Spinners/SmallSpinner'

import './ProductCard.scss'

export const ProductCard = ({
	id,
	name,
	picture_xlarge,
	description,
	price,
}: Product): JSX.Element => {
	const { currentItemId, status } = useTypedSelector(state => state.cart)
	const { addToCart } = useActions()

	return (
		<article
			tabIndex={0}
			aria-label={`sneakers ${name}, unit price ${price} $`}
			className='product'
		>
			<header>
				<img height={200} width={200} src={picture_xlarge} alt={name} />
			</header>

			<section className='product-descriptor'>
				<h2>{name}</h2>
				<p>{description}</p>
			</section>

			<footer>
				{price} $
				<button
					aria-label={`Add to cart sneakers ${name}`}
					onClick={() => addToCart(id)}
				>
					{status === 'loading' && currentItemId === id ? (
						<SmallSpinner />
					) : (
						'Add to cart'
					)}
				</button>
			</footer>
		</article>
	)
}
