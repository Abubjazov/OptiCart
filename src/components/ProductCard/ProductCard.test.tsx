import { render, screen } from '@testing-library/react'

import { ProductCard } from './ProductCard'

const product = {
	name: 'Air 1',
	picture: 'https://i.ibb.co/Y281B6p/nike9.webp',
	description: 'Qui animi et ullam dignissimos.',
	price: 12.32,
}

test('renders ProductCard', () => {
	render(<ProductCard {...product} />)
	const linkElement = screen.getByText(/Air 1/i)
	expect(linkElement).toBeInTheDocument()
})
