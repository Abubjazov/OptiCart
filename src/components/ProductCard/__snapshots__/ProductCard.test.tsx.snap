// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ProductCard should render ProductCard 1`] = `
<DocumentFragment>
  <article
    class="product"
  >
    <header>
      <img
        alt="Paco"
        height="200"
        src="https://i.ibb.co/tDLC3HD/4-200-200.webp"
        width="200"
      />
    </header>
    <section
      class="product-descriptor"
    >
      <h2>
        Paco
      </h2>
      <p>
        Ab odit numquam consectetur aut.
      </p>
    </section>
    <footer>
      54.5 $
      <button>
        Add to cart
      </button>
    </footer>
  </article>
</DocumentFragment>
`;
